### ISSUE TITLE
Bug/Proposal: Support static methods for explicitly defined resolvers

### ISSUE THREAD

**Issue by klairetan** [graphql-python/graphene#1038]
There currently seems to be three ways for a field's resolver function to be determined.

1) We explicitly pass in a resolver function as the `resolver` parameter to the field.
2) We define an instance method on the class `def resolve_${fieldName}` and Graphene will automatically find this to be the resolve fxn.
3) We don't define a function, and Graphene will attempt to use the default resolver function. If the parent object is a dictionary, Graphene will check if the field is a key on the dictionary. Otherwise, it will check if the field is an attribute on the parent object

While #2 is convenient, there could be issues at scale. If an engineer is unfamiliar with the magic Graphene is doing for us behind the scenes, they may change the name of the field from `originalFieldName` to `newFieldName` without realizing that this will break the field since the `resolve_${originalFieldName}` function will no longer apply to this field. 

Therefore, when using Graphene in shared codebases, it seems approach #1 (explicitly passing in a field's resolver function) would be more preferable. However, in most Graphene examples of approach #1, the resolver is always a top-level defined function, such as:

```
import graphene

 def resolve_full_name(parent, info):
        return f"{parent.first_name} {parent.last_name}"

class Person(graphene.ObjectType):
    full_name = graphene.String(name='fullName', resolver=resolve_full_name)
```

This approach works but raises another issue. When the number of fields on an object grows, so does the number of top-level resolver functions that need to be defined and the namespace becomes polluted.

Therefore, it seems it'd be really beneficial if we were able to pass static methods on the class that represents the parent object as resolver functions to the GraphQL parent object's fields, such as:

```
import graphene

class Person(graphene.ObjectType):
     def resolve_full_name(parent, info):
        return f"{parent.first_name} {parent.last_name}"

    full_name = graphene.String(name='fullName', resolver=resolve_full_name)
```

However, the issue with this is that the resolve_full_name gets passed from within the class. And instead of being passed a callable, it's passed the StaticMethod object itself. 

Currently, I'm getting around this through a decorator such as this:

```
import graphene

def graphqlresolver(static_resolver):
    return static_resolver.__func__

class Person(graphene.ObjectType):
     @graphqlresolver
     def resolve_full_name(parent, info):
        return f"{parent.first_name} {parent.last_name}"

    full_name = graphene.String(name='fullName', resolver=resolve_full_name)
```

I'm wondering if anyone has had the same concerns about resolvers and if they've run into these same issues before? If so, I'd love to hear what are solutions you've come up with!

This isn't a bug persay, but it'd be amazing if Graphene could support resolver functions that come from static methods on the same Class. It seems reasonable that this is how a larger team would like to organize their resolvers and fields. 

Basically, I can **go ahead and open a PR** supporting this! Thoughts?

**Comment by jkimbo** [comment]
@klairetan sorry for not getting back to you on this. So there is a suggestion for a new decorator in another issue (https://github.com/graphql-python/graphene/issues/898#issuecomment-493805109) that I think would solve the issues you've described. Basically, instead of defining the field and resolver separately you would define both at the same time with a `@resolves` decorator. Something like this:

```python
class User(ObjectType):
	@resolves(String, required=True)
	def full_name(parent, info):
		return f"{parent.first_name} {parent.last_name}"
```

(There is also contains a code snippet that should mean you can start using it right away)

I think this way of defining resolvers makes a lot of sense and my intention is to try and get it merged into the graphene codebase (though I think I'll call it `@field` insteads of `@resolves`). If you want to try using it please do and let me know if you come across any issues with it.
